enumerable_methods.rb

```diff
@@ -0,0 +1,100 @@
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < length
+        yield(self[i])
+        i += 1
+      end
+    else
+      'No block given'
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < length
+        yield(self[i], i)
+        i += 1
+      end
+    else
+      'No block given'
+    end
+  end
+
+  def my_select
+    ary = []
+    if block_given?
+      my_each { |elem| ary.push(elem) if yield(elem) }
+    else
+      'No block given'
+    end
+    ary
+  end
+
+  def my_all?
+    if block_given?
+      my_each { |elem| return false unless yield(elem) }
+    else
+      my_each { |elem| return false unless elem }
+    end
+    true
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ]  when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern